packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -136,21 +128,21 @@ class InputDateTimePicker extends React.Component {
 	}
 
 	onChange(event, origin) {
-		const { errorMessage, datetime, textInput } = this.state;
+		const { errorMessage, datetime, textInput, errors, previousErrors } = this.state;
 
-		const errorUpdated = errorMessage !== this.state.errorMessage;
+		const errorUpdated = JSON.stringify(previousErrors) !== JSON.stringify(errors);
```

Comment:
This is not very performant.
Please create a function `haveErrorsChanged(oldErrors, newErrors)`
- create js Sets of error messages
- check their size
- check that every oldError message is in newError messages

And so in the next line, you can write `(this.dateHasChanged() || haveErrorsChanged(previousErrors, errors))`, so it's not evaluated if date has changed